app/services/imports/lettings_logs_field_import_service.rb

```diff
@@ -203,5 +205,103 @@ def update_reason(xml_doc)
         @logger.warn("lettings log with old id #{old_id} not found")
       end
     end
+
+    def update_person_details(xml_doc)
+      return if meta_field_value(xml_doc, "form-name").include?("Sales")
+
+      old_id = meta_field_value(xml_doc, "document-id")
+      record = LettingsLog.find_by(old_id:)
+
+      return @logger.warn("lettings log with old id #{old_id} not found") unless record
+      return @logger.info("lettings log #{record.id} has no hhmemb, skipping update") unless record.hhmemb
+
+      if (2..record.hhmemb).all? { |person_index| record.has_any_person_details?(person_index) || record["details_known_#{person_index}"] == 1 }
+        return @logger.info("lettings log #{record.id} has all household member details, skipping update")
+      end
+      if record.hhmemb == 8 || (record.hhmemb + 1..8).none? { |person_index| file_contains_person_details?(xml_doc, person_index) }
+        return @logger.info("lettings log #{record.id} has no additional household member details, skipping update")
+      end
+
+      person_index = 2
+      next_person_index = person_index + 1
+
+      while person_index <= record.hhmemb
```

Comment:
GREAT loop but I feel like this might benefit from a brief comment explaining its approach, took me a while to work out what it's doing as it's a bit _clever_

Reply:
I extracted some methods does it look any clearer now?

Reply:
Yeah much nicer, makes me think you could also extract record["details_known_#{person_index}"] == 1 as a method called details_not_known_for_person(index) so it's clearer that 1 == No

Reply:
Yup, nice!